### PR TITLE
Humans enhancement

### PR DESCRIPTION
**Pull request by jmbau**
Hereunder a mission file to see what is now possible:
[humans_test.mis.txt](https://github.com/SearchAndRescue2/sar2/files/12329385/humans_test.mis.txt)

### CODE REVIEW

File: src/humanio.c [hunk 1]
```diff
@@ -184,18 +194,48 @@ int SARHumanLoadFromFile(sar_human_data_struct *hd, const char *filename)
...
 	    else if(!strcasecmp(buf, "AssistingHumans"))
 	    {
 		FGetValuesF(fp, value, 1);
 		if(hdp_ptr != NULL)
 		{
-		    hdp_ptr->assisting_humans = MAX((int)value[0], 0);
+		    hdp_ptr->assisting_humans =
+		    (
+			(MAX((int)value[0], 0) < SAR_ASSISTING_HUMANS_MAX)
+			? MAX((int)value[0], 0) : SAR_ASSISTING_HUMANS_MAX
```
Comment (hsanjuan) [inline src/humanio.c:237]: There is a `CLIP` function for this, if I'm understanding right. `CLIP((int)value[0], 0, SAR_ASSISTING_HUMANS)`

File: src/humanio.c [hunk 2]
```diff
@@ -184,18 +194,48 @@ int SARHumanLoadFromFile(sar_human_data_struct *hd, const char *filename)
...
 		if(hdp_ptr != NULL)
 		{
 		    hdp_ptr->mass = (float)MAX(value[0], 0.0);	/* kg */
 		}
 	    }
+	    /* Gender */
+	    else if(!strcasecmp(buf, "Gender"))
+	    {
+		/* Male is default gender.
+		* If Female, a breast will be drawn on torso.
```
Comment (hsanjuan) [inline src/humanio.c:214]: Hopefully two breasts?

File: src/simop.c [hunk 3]
```diff
@@ -3409,54 +3409,54 @@ void SARSimApplyGCTL(sar_core_struct *core_ptr, sar_object_struct *obj_ptr)
...
-				hoist->rope_cur = hoist->rope_max;
-
-			    /* Do not update rope_cur_vis, it will be updated
-			    * in the other simulation calls
-			    */
-			}
+			/* Reset hoist values */
+			hoist->rope_cur = hoist->contact_z_max;
+			hoist->rope_cur_vis = hoist->rope_cur;
+			hoist->on_ground = 0;
```
Comment (hsanjuan) [inline src/simop.c:3444]: Is something changing here or just indentation issue?

File: tools/v3dconverter/readme.txt [hunk 4]
```diff
@@ -0,0 +1,40 @@
+
```
Comment (hsanjuan) [inline tools/v3dconverter/readme.txt:1]: Can we put this on a separate PR? Seems out the "human improvement" scope

**Review by hsanjuan — CHANGES_REQUESTED** [review]
Thanks, see comments.

**Comment by hsanjuan** [comment]
> Hereunder a mission file to see what is now possible: [humans_test.mis.txt](https://github.com/SearchAndRescue2/sar2/files/12329385/humans_test.mis.txt)

Perhaps we could put this as part of the test_world scenery? There's not much space (only a boat and a burning platform, but maybe a good excuse to add some land.

File: tools/v3dconverter/readme.txt (hunk 4 above)
Comment (jmbau) [reply tools/v3dconverter/readme.txt:1]: Lost in git, **ONE MORE TIME**...
I don't understand why it is in humans enhancement PR, I pushed this 2 days ago in the v3dconverter branch, but I didn't asked any PR for this branch because even if v3dconverter works (I did all Corsica scenery update new models with it), as written in readme.txt/FOREWORD: it's not yet finished !
How can I remove these files ? With .../Delete file ???

File: src/simop.c (hunk 3 above)
Comment (jmbau) [reply src/simop.c:3444]: Only indentation, no changes here.

File: src/humanio.c (hunk 1 above)
Comment (jmbau) [reply src/humanio.c]: You're right, done!

File: src/humanio.c (hunk 2 above)
Comment (jmbau) [reply src/humanio.c]: Yes, hopefully !  French <-> English translation issue... Done !

File: tools/v3dconverter/readme.txt (hunk 4 above)
Comment (hsanjuan) [reply tools/v3dconverter/readme.txt:1]: It seems you pushed onto this branch too.

First check that the v3dconverter stuff is correctly committed in the other branch.

Then you can do `git rebase -i master` on this branch and select which commits to drop. I can do it for you too.

File: src/human.h [hunk 5]
```diff
@@ -38,12 +37,15 @@ typedef struct {
...
+	/* For gender SAR_HUMAN_FLAG_GENDER_FEMALE (see obj.h). */
+	sar_obj_flags_t preset_entry_flags;
+	
 	/* Color palette */
 	sar_color_struct	color[SAR_HUMAN_COLORS_MAX];
 
 	/* Number of assisting humans (0 for none) */
 	int		assisting_humans;
-
+
```
Comment (hsanjuan) [inline src/human.h:48]:
```suggestion

```

File: src/humanio.c [hunk 6]
```diff
@@ -135,18 +143,18 @@ int SARHumanLoadFromFile(sar_human_data_struct *hd, const char *filename)
...
-		/* Ignore */
+		version_major = (int)value[0];
+		version_minor = (int)value[1];
 	    }
 	    /* PresetAdd */
 	    else if(!strcasecmp(buf, "PresetAdd"))
 	    {
 		char *s = FGetString(fp);
-
+
```
Comment (hsanjuan) [inline src/humanio.c:157]:
```suggestion

```

File: src/objio.c [hunk 7]
```diff
@@ -1696,9 +1698,9 @@ int SARObjLoadSmoke(
 	  case 2:	/* Dark */
 	    tex_name = SAR_STD_TEXNAME_SMOKE_DARK;
 	    break;
-	  case 3:	/* Orange */
-	    tex_name = SAR_STD_TEXNAME_SMOKE_ORANGE;
-	    break;
+      case 3:	/* Orange */
+        tex_name = SAR_STD_TEXNAME_SMOKE_ORANGE;
+        break;
```
Comment (hsanjuan) [inline src/objio.c:1703]:
```suggestion
	 case 3:	/* Orange */
	    tex_name = SAR_STD_TEXNAME_SMOKE_ORANGE;
	    break;
```

File: tools/v3dconverter/readme.txt (hunk 4 above)
Comment (jmbau) [reply tools/v3dconverter/readme.txt:1]: If you can do it, I think that it will be less dangerous for the SarII project... This morning I read some documentation about rebase, but frankly speaking, it is not clear for me.

**Review by hsanjuan — APPROVED** [review]
